share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -316,3 +316,32 @@ tutorial-develop-build:
   needs:
     - artifacts: True
       job: tutorial-develop-generate
+
+########################################
+# Python
+########################################
+.python:
+  variables:
+    SPACK_CI_STACK_NAME: python
+
+python-pr-generate:
+  extends: [ ".python", ".pr-generate"]
+
+python-develop-generate:
+  extends: [ ".python", ".develop-generate"]
+
+python-pr-build:
+  extends: [ ".python", ".pr-build" ]
+  trigger:
+    include:
+      - artifact: jobs_scratch_dir/cloud-ci-pipeline.yml
+        job: python-pr-generate
+    strategy: depend
```

Comment:
To allow this job to run as soon as the corresponding generate has finished (and not wait until the entire stage is finished, which is the default), we need a `needs` here like [this](https://github.com/spack/spack/blob/f81d84dfc6b1c3a821dffab785e62ecd79f130ab/share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml#L124) one.